pkg/controller/common.go

```diff
@@ -2177,3 +2181,16 @@ func (c *SharedState) getNodesForReplica(ctx context.Context, volumeName string,
 
 	return filtered, nil
 }
+
+func verifyDeleted(obj interface{}, objectDeleted bool) (bool, error) {
```

Comment:
Consider `verifyAzVolumeAttachmentDeleted` since this is specific to that object.